packages/qwik/src/core/util/case.unit.ts

```diff
@@ -2,7 +2,7 @@ import { assert, test } from 'vitest';
 import { fromCamelToKebabCase } from './case';
 
 test('should convert to kebab', () => {
-  assert.equal(fromCamelToKebabCase('HelloWorld'), '-hello-world');
+  assert.equal(fromCamelToKebabCase('HelloWorld'), 'hello-world');
   assert.equal(fromCamelToKebabCase('on:ClicK'), 'on:-clic-k');
```

Comment:
Shouldn't you handle this case then too? `.replace(/(^|:)-/, '')`?

All in all though, we're doing extra work for no real benefit. I think it's better to leave the kebabcase as-is.
There might even be someone that sends two different events `myEvent` and `MyEvent`, and this change would break that.